.ci/Jenkinsfile

```diff
@@ -330,131 +436,106 @@ def generateFunctionalTestStep(Map args = [:]){
     workerLabels = "centos-8-aarch64"
   }
 
-  if (platformLabels.contains("windows")) {
-    // For windows testing we need to make sure that the host executor is a linux machine and nest the windows withNode within that
-    workerLabels = "ubuntu-18.04"
-    tags += " && ~@skip:windows"
-  }
-
   // Skip scenarios per platform
   tags += " && ~@skip:${goArch}"
 
   // Setup environment for platform
   def envContext = []
-  envContext.add("GOARCH=${goArch}")
   envContext.add("PROVIDER=${platformProvider}")
+  envContext.add("SUITE=${suite}")
   envContext.add("ELASTIC_APM_GLOBAL_LABELS=branch_name=${BRANCH_NAME},build_pr=${isPR()},build_id=${env.BUILD_ID},go_arch=${goArch},beat_version=${env.BEAT_VERSION},stack_version=${env.STACK_VERSION}")
 
-  return {
-    withNode(labels: "${workerLabels}", sleepMax: 20, forceWorkspace: true){
-      deleteDir()
-      unstash 'source'
+  if (testRunner){
+    return {
       try {
-        withDockerEnv(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}") {
-          filebeat(output: "docker_logs_${platformLabels}_${goArch}_${suite}_${name}.log", workdir: "${env.WORKSPACE}") {
-            dir("${BASE_DIR}") {
-              // Doing a windows agent test, start up docker before the new environment is applied for windows
-              if (platformLabels.contains("windows")) {
-                sh script: "docker-compose -f cli/config/compose/profiles/${suite}/docker-compose.yml up -d", label: "Deploying ${suite} via docker"
-                linuxIp = grabWorkerIP()
-                def mingwArch = is32() ? '32' : '64'
-                def chocoPath = 'C:\\ProgramData\\chocolatey\\bin'
-                def chocoPython3Path = 'C:\\Python38;C:\\Python38\\Scripts'
-                path = "${env.WORKSPACE}\\bin;${chocoPath};${chocoPython3Path};C:\\tools\\mingw${mingwArch}\\bin;C:\\windows\\system32;C:\\windows\\system32\\WindowsPowerShell\\v1.0\\"
-                envContext.add("KIBANA_URL=https://${linuxIp}:5601")
-                envContext.add("ELASTICSEARCH_URL=https://${linuxIp}:9200")
-                envContext.add("FLEET_URL=https://${linuxIp}:8220")
-                envContext.add("PROVIDER=remote") // a remote provider is needed for Windows
-                envContext.add("OP_LOG_LEVEL=TRACE")
-                envContext.add("PATH=${path}")
-              }
-              withEnv(envContext) {
+        dir("${BASE_DIR}") {
+          withEnv(envContext) {
+            // This step will help to send the APM traces to the
+            // APM service defined by the Otel Jenkins plugin.
+            withOtelEnv() {
```

Comment:
We must make sure the OTEL variables are sent to the testRunner machine, as I guess this wrapper won't work for the remote SSH connection. Let's think about the security implications of sending those credentials to the machines 🤔 

With that in mind, if we are not sending APM traces it could be OK until we resolve this concern

Reply:
Just to elaborate, one of the ideas is to make vault accessible from within those test runners to get the necessary info to send to APM